### PR TITLE
Fix State being mutated and move next step value to Util Function

### PR DESCRIPTION
**Pull request by bhattaraib58**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -49,42 +50,33 @@ export const NumberScroller: FC<NumberScrollerProps> = ({
...
-    } else {
-      return (currentNumber -= Math.min(
-        Math.abs(step),
-        (to ?? 0) + currentNumber
-      ));
-    }
-  };
   useEffect(() => {
-    const runEngine = () => {
+    const runEngine = currentNumber => {
```
Comment (sbimochan) [inline src/index.tsx:59]: Put type for current number

**Comment by sbimochan** [comment]
Amazing! Thank you for your contribution.

File: src/index.tsx [hunk 2]
```diff
@@ -49,42 +50,33 @@ export const NumberScroller: FC<NumberScrollerProps> = ({
   toLocaleStringProps,
   step = 1,
   timeout = 1000,
-  to,
+  to = 0,
```
Comment (KevinVandy) [inline src/index.tsx:53]: If we default the "to" prop to 0, that means that the "fallback" prop will never get used, right? Might as well remove the fallback prop all together?

File: src/utils.ts [hunk 3]
```diff
@@ -10,3 +10,30 @@ export function getRoundedFloatWithPrecision(num: number, precision = 2) {
 
   return Math.round((num + Number.EPSILON) * ROUND_UPTO) / ROUND_UPTO;
 }
+
+/**
+ * Get Next Step Value for the number change.
+ *
```
Comment (KevinVandy) [inline src/utils.ts:16]: I'm really curious. What is the benefit to having both typescript types and jsdoc stuff. Do they help different IDEs? Is this redundant?

Comment (sbimochan) [reply src/utils.ts:16]: you're right. removing it next

File: src/index.tsx (hunk 2 above)
Comment (sbimochan) [reply src/index.tsx:53]: right.

**Pull request merged**